### ISSUE TITLE
Classifier-Free Guidance Formulation

### ISSUE THREAD

**Issue by MoayedHajiAli** [lucidrains/DALLE2-pytorch#303]
In the classifier-free guidance paper, the formulation is as follow
![image](https://github.com/lucidrains/DALLE2-pytorch/assets/52598644/85761e78-c71a-474e-86c7-fe56d83ba28e)

However, it is implemented in DALL-2 as 
`null_logits + (logits - null_logits) * cond_scale`
According to the formula it should be
`logits + (logits - null_logits) * cond_scale`
where the logits are the conditional score and null_logits are the unconditional ones. 

Can you please clarify the reason for this modification? 
Thank you!

**Comment by nshidqi** [comment]
@MoayedHajiAli Have you found the reason?
I'm also wondering why

**Comment by tengteng95** [comment]
@nshidqi @MoayedHajiAli  I also have the same question. May I ask if you have found the reason now?

**Comment by MoayedHajiAli** [comment]
Hi @tengteng95. I couldn't find the reason
